subsys/bluetooth/mesh/net.c

```diff
@@ -803,7 +803,15 @@ void bt_mesh_net_recv(struct net_buf_simple *data, int8_t rssi,
 
 static void ivu_refresh(struct k_work *work)
 {
-	bt_mesh.ivu_duration += BT_MESH_IVU_HOURS;
+	uint8_t duration = bt_mesh.ivu_duration;
+
+	duration += BT_MESH_IVU_HOURS;
+
+	if (duration < bt_mesh.ivu_duration) {
+		duration = BT_MESH_IVU_MAX_COUNTER;
+	}
+
+	bt_mesh.ivu_duration = duration;
```

Comment:
Why not just `bt_mesh.ivu_duration = MIN(UNIT8_MAX, bt_mesh.ivu_duration + BT_MESH_IVU_HOURS);`?